src/troubleshooting/content.md

```diff
@@ -0,0 +1,641 @@
+    }
+  ]
+}
+```
+
+In practice, you should look at these lines:
+
+```yaml|highlight=4,7,21,22|title=hello-world.yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: app
+spec:
+  ports:
+  - port: 80
+    targetPort: 8080
+  selector:
+    name: app
+---
+apiVersion: networking.k8s.io/v1beta1
+kind: Ingress
+metadata:
+  name: app
+spec:
+  rules:
+  - http:
+    paths:
+    - backend:
+        serviceName: app
+        servicePort: 80
+      path: /
+```
+
+_How do you test that the Ingress works?_
+
+You can use the same strategy as before with `kubectl port-forward`, but instead of connecting to a service, you should connect to the Ingress controller.
+
+First, retrieve the Pod name for the Ingress controller with:
+
+```terminal|command=1|title=bash
+kubectl get pods --all-namespaces
+NAMESPACE   NAME                              READY STATUS
+kube-system coredns-5644d7b6d9-jn7cq          1/1   Running
+kube-system etcd-minikube                     1/1   Running
+kube-system kube-apiserver-minikube           1/1   Running
+kube-system kube-controller-manager-minikube  1/1   Running
+kube-system kube-proxy-zvf2h                  1/1   Running
+kube-system kube-scheduler-minikube           1/1   Running
+kube-system nginx-ingress-controller-6fc5bcc  1/1   Running
+```
+
+Identify the Ingress Pod (which might be in a different Namespace) and describe it to retrieve the port:
+
+```terminal|command=1|title=bash
+kubectl describe pod nginx-ingress-controller-6fc5bcc
+ --namespace kube-system \
+ | grep Ports
+```
+
+Finally, connect to the Pod:
+
+```terminal|command=1|title=bash
+kubectl port-forward nginx-ingress-controller-6fc5bcc 3000:80 --namespace kube-system
+```
+
+At this point, every time you visit port 3000 on your computer, the request is forwarded to port 80 on the Ingress controller Pod.
+
+If you visit <http://localhost:3000>, you should find the app serving a web page.
+
+## Recap on ports
+
+Here's a quick recap on what ports and labels should match:
+
+1. The Service selector should match the label of the Pod
+1. The Service `targetPort` should match the `containerPort` of the container inside the Pod
+1. The Service port can be any number. Multiple Services can use the same port because they have different IP addresses assigned.
+1. The `servicePort` of the Ingress should match the `port` in the Service
+1. The name of the Service should match the field `serviceName` in the Ingress
+
+Knowing how to structure your YAML definition is only part of the story.
+
+_What happens when something goes wrong?_
+
+Perhaps the Pod doesn't start, or it's crashing.
+
+## 3 steps to troubleshoot Kubernetes deployments
+
+It's essential to have a well defined mental model of how Kubernetes works before diving into debugging a broken deployment.
+
+Since there're three components in every deployment, you should debug all of them in order, starting from the bottom.
```

Comment:
as far as i can tell, people usually writing out "there are", "there're" feels weird to me, but I'm not native.

Reply:
This goes back to the chat we had.
It's **not** important to please the native speakers but to use a simple language. Not all reader are natives.
Changed.